### PR TITLE
Jacksonise and Cleanup commit logs + Some Task API simplification

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/tasks/mock/MockBackgroundTask.java [hunk 1]
```diff
@@ -88,12 +88,17 @@ public static void clearTasks() {
...
+            throw new RuntimeException("Invalid task config");
+        }
+
         Json id = json.at("id");
         if (id == null) {
             LOG.error("Missing id in {}: {}", this.getClass().getName(), json);
         }
-        this.id = TaskId.of(id.asString());
+        this.id = TaskId.generate();
+        //this.id = TaskId.of(id.asString());
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/tasks/mock/MockBackgroundTask.java]: 🐐

File: grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerInsertTest.java [hunk 2]
```diff
@@ -224,7 +234,7 @@ public void POSTGraqlInsertAndNoLogs_PPTaskIsNotSubmitted(){
 
         sendRequest(query);
 
-        verify(taskManager, times(1)).addTask(any(), any());
+        verify(taskManager, times(0)).addTask(any(), any());
```
Comment (aelred) [inline grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerInsertTest.java]: Whoops, were these tests wrong all along? 😱

File: grakn-kb/src/main/java/ai/grakn/factory/GraknSessionImpl.java [hunk 3]
```diff
@@ -93,10 +95,7 @@
             ThreadFactory namedThreadFactory = new ThreadFactoryBuilder()
                     .setNameFormat("commit-log-subbmit-%d").build();
             commitLogSubmitter = Executors.newSingleThreadScheduledExecutor(namedThreadFactory);
-            commitLogSubmitter.scheduleAtFixedRate(() -> {
-                submitLogs(tx);
-                submitLogs(txBatch);
-            }, 0, LOG_SUBMISSION_PERIOD, TimeUnit.SECONDS);
+            commitLogSubmitter.scheduleAtFixedRate(this::submitLogs, 0, LOG_SUBMISSION_PERIOD, TimeUnit.SECONDS);
```
Comment (aelred) [inline grakn-kb/src/main/java/ai/grakn/factory/GraknSessionImpl.java]: When/how do we submit the commit logs for batch transactions?

Comment (fppt) [reply grakn-kb/src/main/java/ai/grakn/factory/GraknSessionImpl.java]: Both the normal logs and batch logs are put in the same `CommitLogHandler` so we only need one submission

File: grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerInsertTest.java (hunk 2 above)
Comment (fppt) [reply grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerInsertTest.java]: Yep. . . They were wrong.

**Comment by fppt** [comment]
@aelred re-review please

File: grakn-engine/src/main/java/ai/grakn/engine/tasks/mock/MockBackgroundTask.java [hunk 4]
```diff
@@ -88,12 +88,16 @@ public static void clearTasks() {
...
+        if(json.equals(Json.object())){
+            throw new RuntimeException("Invalid task config");
+        }
+
         Json id = json.at("id");
         if (id == null) {
             LOG.error("Missing id in {}: {}", this.getClass().getName(), json);
         }
-        this.id = TaskId.of(id.asString());
+        this.id = TaskId.generate();
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/tasks/mock/MockBackgroundTask.java]: It looks like we ignore the ID from the JSON configuration here. Is that deliberate? I can't remember how any of this task stuff works anymore.

Comment (fppt) [reply grakn-engine/src/main/java/ai/grakn/engine/tasks/mock/MockBackgroundTask.java]: The Task ID is kind of in limbo right now. It's not actually not used anywhere anymore properly. The old property tests use it in weird ways, hence the weird mock above.

I could get rid of the old property tests but I am not 100% sure yet if I should.

These odd mocks should die in another PR though. They probably made sense at some point but not anymore

**Pull request merged**